nBTC/sources/nbtc.move

```diff
@@ -1,8 +1,126 @@
 /// Module: nbtc
 module nbtc::nbtc;
 
+use sui::coin::{Self, Coin, TreasuryCap};
+use sui::url;
+use bitcoin_spv::light_client::LightClient;
+use bitcoin_spv::transaction::make_transaction;
+use bitcoin_spv::light_client::prove_payment;
+use sui::table::{Self, Table};
+use sui::address;
 
-// For Move coding conventions, see
-// https://docs.sui.io/concepts/sui-move-concepts/conventions
+/// CONSTANTS
+/// --COIN METADATA--
+const DECIMALS: u8 = 8;
+const SYMBOL: vector<u8> = b"NBTC";
+const NAME: vector<u8> = b"Native BTC";
+const DESCRIPTION: vector<u8> = b"Natvie synthetic BTC powered by IKA.";
+const ICON_URL: vector<u8> = b"icon.url";
+/// --CONFIG--
+const TRUSTED_LIGHT_CLIENT_ID: address =@0xCA;
+const FALLBACK_ADDRESS: address = @0xCF;
+const DWALLET: vector<u8> = b"btc_address"; 
 
 
+/// OTW
+public struct NBTC has drop {}
+
+/// ERRROS
+    const ETxAlreadyUsed: u64 = 0;
+    const EMintAmountIsZero: u64 = 1;
+    const EUntrustedLightClient: u64 = 2;
+
+
+/// STRUCTS
+public struct WrappedTreasuryCap has key, store {
+    id: UID,
+    cap: TreasuryCap<NBTC>,
+    tx_ids: Table<vector<u8>, bool>,
+    trusted_lc_id: ID,
+    fallback_address: address,
+    dwallet: vector<u8>,
+}
+
+fun init(witness: NBTC, ctx: &mut TxContext) {
+    let (treasury_cap, metadata) = coin::create_currency<NBTC>(
+        witness,
+        DECIMALS,
+        SYMBOL,
+        NAME,
+        DESCRIPTION,
+        option::some(url::new_unsafe_from_bytes(ICON_URL)),
+        ctx
+    );
+
+    transfer::public_freeze_object(metadata);
+    let treasury =
+     WrappedTreasuryCap {
+         id: object::new(ctx),
+         cap: treasury_cap, 
+         tx_ids: table::new<vector<u8>, bool>(ctx),
```

Comment:
I think we should use dynamic fields, as we only check whether they exist or not.